doc/source/user_guide/duplicates.rst

```diff
@@ -0,0 +1,172 @@
+
+   df2
+   df2.index.is_unique
+   df2.columns.is_unique
+
+.. note::
+
+   Checking whether an index is unique is somewhat expensive for large datasets.
+   Pandas does cache this result, so re-checking on the same index is very fast.
+
+:meth:`Index.duplicated` will return a boolean ndarray indicating whether a
+label is a repeat.
+
+.. ipython:: python
+
+   df2.index.duplicated()
+
+Which can be used as a boolean filter to drop duplicate rows.
+
+.. ipython:: python
+
+   df2.loc[~df2.index.duplicated(), :]
+
+If you need additional logic to handle duplicate labels, rather than just
+dropping the repeats, using :meth:`~DataFrame.groupby` on the index is a common
+trick. For example, we'll resolve duplicates by taking the average of all rows
+with the same label.
+
+.. ipython:: python
+
+   df2.groupby(level=0).mean()
+
+.. _duplicates.disallow:
+
+Disallowing Duplicate Labels
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. versionadded:: 1.1.0
+
+As noted above, handling duplicates is an important feature when reading in raw
+data. That said, you may want to avoid introducing duplicates as part of a data
+processing pipeline (from methods like :meth:`pandas.concat`,
+:meth:`~DataFrame.rename`, etc.). Both :class:`Series` and :class:`DataFrame`
+can be created with the argument ``allows_duplicate_labels=False`` to *disallow*
+duplicate labels (the default is to allow them). If there are duplicate labels,
+an exception will be raised.
+
+.. ipython:: python
+   :okexcept:
+
+   pd.Series([0, 1, 2], index=['a', 'b', 'b'], allows_duplicate_labels=False)
+
+This applies to both row and column labels for a :class:`DataFrame`
+
+.. ipython:: python
+   :okexcept:
+
+   pd.DataFrame([[0, 1, 2], [3, 4, 5]], columns=["A", "B", "C"],
+                allows_duplicate_labels=False)
+
+This attribute can be checked or set with :attr:`~DataFrame.allows_duplicate_labels`,
+which indicates whether that object can have duplicate labels.
+
+.. ipython:: python
+
+   df = pd.DataFrame({"A": [0, 1, 2, 3]},
+                     index=['x', 'y', 'X', 'Y'],
+                     allows_duplicate_labels=False)
+   df
+   df.allows_duplicate_labels
+
+Performing an operation that introduces duplicate labels on a ``Series`` or
+``DataFrame`` that disallows duplicates will raise an
+:class:`errors.DuplicateLabelError`.
+
+.. ipython:: python
+   :okexcept:
+
+   df.rename(str.upper)
+
+Duplicate Label Propagation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In general, disallowing duplicates is "sticky". It's preserved through
+operations.
+
+.. ipython:: python
+   :okexcept:
+
+   s1 = pd.Series(0, index=['a', 'b'], allows_duplicate_labels=False)
+   s1
+   s1.head().rename({"a": "b"})
+
+.. warning::
+
+   This is an experimental feature. Currently, many methods fail to
+   propagate the ``allows_duplicate_labels`` value. In future versions
+   it is expected that every method taking or returning one or more
+   DataFrame or Series objects will propagate ``allows_duplicate_labels``.
```

Comment:
maybe create a master tracker issue with a list of known methods that don't propagate and link to the issue here.

Reply:
I'm always a bit hesitant to link "permanent" things (like the docs for a specific version) back to mutable things like a GH issue. But I agree that something like this might be useful for users.

For *us developers*, the authoritative list is probably all the not_implemented marks in https://github.com/pandas-dev/pandas/blob/a07748f93259735d6af4e80beee21b76414d6689/pandas/tests/generic/test_finalize.py#L32.